Utilizr/Util/DomainHelper.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Utilizr.Util
+{
+    public static class DomainHelper
+    {
+        public static string GetSubDomain(string fullurl)
+        {
+            string name = fullurl.StartsWith("http://") || fullurl.StartsWith("https://") ? fullurl : $"http://{fullurl}";
+            var url = new Uri(name);
+
+            if (url.HostNameType == UriHostNameType.Dns)
+            {
+                string host = url.Host;
+
+                var nodes = host.Split('.');
+
+                if (nodes.Length > 2)
+                {
+                    return string.Format("{0}.{1}", nodes[nodes.Length - 2], nodes[nodes.Length - 1]);
+                }
+                return null;
+            }
+
+            return null;
+        }
+
+        public static string GetDomain(string url)
+        {
+            string name = url.StartsWith("http://") || url.StartsWith("https://") ? url : $"http://{url}";
+            var uri = new Uri(name);
+            return uri.Host;
+        }
```

Comment:
- We can refactor stripping out the protocols to a single function to avoid duplication.
- Inconsistent return types, `null` and `string` instead of `string?`.
- I don't know if we need the `UriHostNameType.Dns`.
- We may need to improve the `GetSubDomain()` logic, I think sites like `test.co.uk` may have issues (this can be in a `[TestCase()]`).